eras/conway/impl/src/Cardano/Ledger/Conway/CDDL.hs

```diff
@@ -0,0 +1,971 @@
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE OverloadedLists #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}
+
+{-# HLINT ignore "Use camelCase" #-}
+{-# HLINT ignore "Evaluate" #-}
+
+module Cardano.Ledger.Conway.CDDL where
+
+import Codec.CBOR.Cuddle.Huddle
+import Data.Function (($))
+import Data.Semigroup ((<>))
+import Data.Text qualified as T
+import Data.Word (Word64)
+import GHC.Float (Double)
+import GHC.Num (Integer)
+import GHC.Show (Show (show))
+
+default (Integer, Double)
+
+conway :: Huddle
+conway = collectFrom [block, transaction]
+
+block :: Rule
+block =
+  "block"
+    =:= arr
+      [ a header
+      , "transaction_bodies" ==> arr [0 <+ a transaction_body]
+      , "transaction_witness_sets"
+          ==> arr [0 <+ a transaction_witness_set]
+      , "auxiliary_data_set"
+          ==> mp [0 <+ asKey transaction_index ==> auxiliary_data]
+      , "invalid_transactions" ==> arr [0 <+ a transaction_index]
+      ]
+
+transaction :: Rule
+transaction =
+  "transaction"
+    =:= arr
+      [ a transaction_body
+      , a transaction_witness_set
+      , a VBool
+      , a (auxiliary_data // VNil)
+      ]
+
+transaction_index :: Rule
+transaction_index = "transaction_index" =:= VUInt `sized` (2 :: Word64)
+
+header :: Rule
+header = "header" =:= arr [a header_body, "body_signature" ==> kes_signature]
+
+header_body :: Rule
+header_body =
+  "header_body"
+    =:= arr
+      [ "block_number" ==> VUInt
+      , "slot" ==> VUInt
+      , "prev_hash" ==> hash32
+      , "issuer_vkey" ==> vkey
+      , "vrf_vkey" ==> vkey
+      , "vrf_result" ==> vrf_cert
+      , "block_body_size" ==> VUInt
+      , "block_body_hash" ==> hash32
+      , a operational_cert
+      , a (arr [a protocol_version])
+      ]
+
+operational_cert :: Rule
+operational_cert =
+  "operational_cert"
+    =:= arr
+      [ "hot_vkey" ==> kes_vkey
+      , "sequence_number" ==> VUInt
+      , "kes_period" ==> VUInt
+      , "sigma" ==> signature
+      ]
+
+protocol_version :: Named Group
+protocol_version = "protocol_version" =:~ grp [a major_protocol_version, a VUInt]
+
+next_major_protocol_version :: Rule
+next_major_protocol_version = "next_major_protocol_version" =:= (10 :: Integer)
+
+major_protocol_version :: Rule
+major_protocol_version = "major_protocol_version" =:= (1 :: Integer)
+
+transaction_body :: Rule
+transaction_body =
+  "transaction_body"
+    =:= mp
+      [ idx 0 ==> set transaction_input
+      , idx 1 ==> arr [0 <+ a transaction_output]
+      , idx 2 ==> coin
+      , opt (idx 3 ==> VUInt)
```

Comment:
We now have defined [`slot_no`](https://github.com/IntersectMBO/cardano-ledger/blob/f4119c9cd27e69943dcf91d22d0f9905be32fa75/eras/conway/impl/cddl-files/conway.cddl#L589)
```suggestion
      , opt (idx 3 ==> slot_no)
```